python/lsst/obs/base/gen2to3/translators.py

```diff
@@ -453,7 +453,8 @@ def addGenericInstrumentRules(self, instrumentName: str,
         self.addRule(CalibKeyHandler(detectorKey), gen2keys=("calibDate",))
 
     def makeMatching(self, datasetTypeName: str, gen2keys: Dict[str, type], instrument: Optional[str] = None,
-                     skyMap: Optional[BaseSkyMap] = None, skyMapName: Optional[str] = None):
+                     skyMap: Optional[BaseSkyMap] = None, skyMapName: Optional[str] = None,
+                     log: Optional[Log] = None):
```

Comment:
I've refactored the logging (including some existing logging in gen2to3) to pass loggers down in `__init__`; that was a nice cleanup, with the only downside being a tiny bit of two-step initialization reduce the number of places I have to forward it (which wasn't too bad, or I would have just done the pass-throughs).  I do like having the loggers here at least be children of the task logger (rather than module-scope loggers) so it's more straightforward to control them all together.